cmd/skopeo/proxy.go

```diff
@@ -568,6 +574,65 @@ func (h *proxyHandler) GetBlob(args []interface{}) (replyBuf, error) {
 	return ret, nil
 }
 
+func (h *proxyHandler) LayerInfoJSON(args []interface{}) (replyBuf, error) {
+  h.lock.Lock()
+	defer h.lock.Unlock()
+
+	var ret replyBuf
+
+	if h.sysctx == nil {
+		return ret, fmt.Errorf("client error: must invoke Initialize")
+	}
+	
+  if len(args) != 1 {
+		return ret, fmt.Errorf("found %d args, expecting (imgid)", len(args))
+	}
+  
+  imgref, err := h.parseImageFromID(args[0])
+  if err != nil {
+    return ret, err
+  }
+
+  ctx := context.TODO()
+
+  var infos []types.BlobInfo
+  infos, err = imgref.src.LayerInfosForCopy(ctx, nil)
+  if err != nil {
+    return ret, err
+  }
+  if infos == nil {
+    rawManifest, _, err := imgref.src.GetManifest(ctx, nil)
+    if err != nil {
+      return ret, err
+    }
+    var serManifest manifest.OCI1
+    if err = json.Unmarshal(rawManifest, &serManifest); err != nil {
+      err = fmt.Errorf("invalid request: %v", err)
+		  return ret, err
+    }
+    layerInfos := serManifest.LayerInfos()
+    for _, layerInfo := range layerInfos {
+      infos = append(infos, layerInfo.BlobInfo)
+    }
+  }
+
+  
+  var serStruct layersInfo
+  serStruct.Success = "successfull"
+  serStruct.Layers = infos
+
+  serialized, err := json.Marshal(&serStruct)
+	if err != nil {
+		return ret, err
+	}
+  
+  return h.returnBytes(nil, serialized)
+  /*
+  ret.value = serialized
+  return ret, err
+  */
```

Comment:
Why is this using the much more costly pipe mechanism instead of just returning the data as in the commented-out path?